### PR TITLE
Patch action synonym implementation to use 'strcmp'

### PR DESCRIPTION
**Pull request by isaacfink21**
PR #736 was implemented incorrectly to use the `==` operator rather than `strcmp` when checking for action synonyms. This PR patches these errors.

### CODE REVIEW

File: src/action_management/src/actionmanagement.c [hunk 1]
```diff
@@ -83,11 +83,11 @@ int do_item_action(chiventure_ctx_t *c, action_type_t *a, item_t *i, char **ret_
     }
 
     /* use representative c_name for action synonyms */
-    if(a->c_name == "PICKUP") 
+    if(strcmp(a->c_name, "PICKUP") == 0)
```
[the remark this answers is not in the document]
Comment (isaacfink21) [reply src/action_management/src/actionmanagement.c]: Ah thank you, this should be fixed now

[the remark this answers is not in the document]
Comment (MaxineK36) [reply src/action_management/src/actionmanagement.c]: This technically works, but ideally these should be whatever the maximum length of an action name is (maybe BUFFER_SIZE?)

**Comment by isaacfink21** [comment]
PR approved! Comparisons in the implementation of action synonyms have been patched from `==` to `strncmp`, and now operate correctly in a demo game.

**Comment by MaxineK36** [comment]
Issue Score: ✔️++

Comments:
Great work! This small PR does a great job of being explicit about changes that were made.

**Pull request merged**